zcash_client_sqlite/src/testing.rs

```diff
@@ -471,8 +471,31 @@ where
         let (res, nf) = self.generate_block_at(
             height,
             prior_cached_block.chain_state.block_hash(),
-            fvk,
-            &[FakeCompactOutput::new(address_type, value)],
+            &[FakeCompactOutput::new(fvk, address_type, value)],
+            prior_cached_block.sapling_end_size,
+            prior_cached_block.orchard_end_size,
+            false,
+        );
+
+        (height, res, nf)
+    }
+
+    /// Creates a fake block at the expected next height containing multiple outputs
+    /// and inserts it into the cache.
+    pub(crate) fn generate_next_block_multi<Fvk: TestFvk>(
+        &mut self,
+        fvk: &Fvk,
+        address_type: AddressType,
+        value: NonNegativeAmount,
+    ) -> (BlockHeight, Cache::InsertResult, Fvk::Nullifier) {
+        let pre_activation_block = CachedBlock::none(self.sapling_activation_height() - 1);
+        let prior_cached_block = self.latest_cached_block().unwrap_or(&pre_activation_block);
+        let height = prior_cached_block.height() + 1;
+
+        let (res, nf) = self.generate_block_at(
+            height,
+            prior_cached_block.chain_state.block_hash(),
+            &[FakeCompactOutput::new(fvk, address_type, value)],
```

Comment:
This method makes no sense to me. Its name (`generate_next_block_multi`) and documentation indicates that it should include multiple outputs, but its code is identical to `generate_next_block`; the code here explicitly only adds one output. Either fix the method to take multiple outputs, or remove it.

EDIT: Looks like this was fixed in the third commit. This fix should definitely be moved into the second commit, as otherwise that commit makes no sense.